src/analyze.cpp

```diff
@@ -2648,6 +2649,29 @@ static Error resolve_enum_zero_bits(CodeGen *g, ZigType *enum_type) {
                     buf_sprintf("consider 'union(enum)' here"));
         }
 
+        AstNode *tag_value = field_node->data.struct_field.value;
+
+        if (buf_eql_str(type_enum_field->name, "_")) {
+            if (decl_node->data.container_decl.init_arg_expr == nullptr) {
+                add_node_error(g, field_node, buf_sprintf("non-exhaustive enum must specify size"));
+                enum_type->data.enumeration.resolve_status = ResolveStatusInvalid;
+            }
+            if (field_count > 1 && log2_u64(field_count - 1) == enum_type->size_in_bits) {
```

Comment:
Isn't this impossible to trigger for a non-`extern` enum? The enum tag size is chosen before the `_` field is skipped.
Moreover the `field_count` is now invalid and `enumeration.fields` contains one entry too many.

Reply:
Tag size must be specified for non-exhaustive enums so it is possible to hit.
```zig
const E = enum(u2) {
    a,
    b,
    c,
    d,
    _, // error: non-exhaustive enum specifies every value
};
```